### ISSUE TITLE
Add suggested auto-complete from Nicknames Cache when typing @...

### ISSUE THREAD

**Issue by EdwardHinkle** [EdwardHinkle/indigenous-ios#84]

**Comment by aaronpk** [comment]
There's a brainstorming issue for Micropub to have a way to query a site's nicknames cache https://indieweb.org/Micropub-brainstorming#Query_for_contacts

**Comment by EdwardHinkle** [comment]
My first thought is concerning potentially writing posts offline. I think I would like to be able to query my micropub endpoint and get an array of contacts, in the same way I can get an array of syndication targets. This allows the ability to be writing a post offline and to power either person tagging or @-mentioning.

Right now you can only have one draft post at a time (if you click cancel while writing a post, it will ask if you want to save you post as a draft, if so it will auto-fill the form when you open it back up again).

Eventually I would like if you are offline to click "Post" and to have it auto-post when it gets internet again. So in those cases I would want to be able to handle all nickname cache stuff on device after the initial login (and maybe periodically refetching the cache every couple of days or something).

I can see how it could be useful to query the server with the string so far in a situation where you know you have internet.

**Comment by aaronpk** [comment]
Yeah the offline case makes sense for sure. I guess i was just worried the list of names might get too big. So maybe the client needs a mechanism to sync the list and keep it updated.

**Comment by EdwardHinkle** [comment]
That's true... it could end up eventually being a big list.
